{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "3a22cd32-0a2f-43de-b054-061726764cb7",
   "metadata": {},
   "source": [
    "### Notebook Purpose\n",
    "\n",
    "**Transform the gene-disease data from the G2P database to include additional mechanistic encoding columns in the form of LoF, GoF, DN, and a confidence score assessing the prediction.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "8b02290b-6aa4-4179-9a05-5a607c76f634",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "d7a97de2-3cbf-41ea-bc5f-949ceb9468d8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>g2p id</th>\n",
       "      <th>gene symbol</th>\n",
       "      <th>gene mim</th>\n",
       "      <th>hgnc id</th>\n",
       "      <th>previous gene symbols</th>\n",
       "      <th>disease name</th>\n",
       "      <th>disease mim</th>\n",
       "      <th>disease MONDO</th>\n",
       "      <th>allelic requirement</th>\n",
       "      <th>cross cutting modifier</th>\n",
       "      <th>...</th>\n",
       "      <th>variant consequence</th>\n",
       "      <th>variant types</th>\n",
       "      <th>molecular mechanism</th>\n",
       "      <th>molecular mechanism categorisation</th>\n",
       "      <th>molecular mechanism evidence</th>\n",
       "      <th>phenotypes</th>\n",
       "      <th>publications</th>\n",
       "      <th>panel</th>\n",
       "      <th>comments</th>\n",
       "      <th>date of last review</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>G2P00001</td>\n",
       "      <td>HMX1</td>\n",
       "      <td>142992.0</td>\n",
       "      <td>5017</td>\n",
       "      <td>H6; NKX5-3</td>\n",
       "      <td>HMX1-related oculoauricular syndrome</td>\n",
       "      <td>612109</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biallelic_autosomal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>absent gene product</td>\n",
       "      <td>NaN</td>\n",
       "      <td>loss of function</td>\n",
       "      <td>inferred</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0000647; HP:0004328; HP:0000639; HP:0000482...</td>\n",
       "      <td>18423520; 25574057; 29140751</td>\n",
       "      <td>DD; Eye</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2019-09-26 16:23:46+00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>G2P00008</td>\n",
       "      <td>YAP1</td>\n",
       "      <td>606608.0</td>\n",
       "      <td>16262</td>\n",
       "      <td>YAP-1; YAP65</td>\n",
       "      <td>YAP1-related coloboma, ocular, with or without...</td>\n",
       "      <td>120433</td>\n",
       "      <td>NaN</td>\n",
       "      <td>monoallelic_autosomal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>absent gene product</td>\n",
       "      <td>NaN</td>\n",
       "      <td>loss of function</td>\n",
       "      <td>inferred</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0000407; HP:0003829; HP:0002006; HP:0001249...</td>\n",
       "      <td>24462371; 27267789</td>\n",
       "      <td>DD; Eye</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2017-05-12 10:48:00+00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>G2P00009</td>\n",
       "      <td>BBS9</td>\n",
       "      <td>607968.0</td>\n",
       "      <td>30000</td>\n",
       "      <td>B1; PTHB1</td>\n",
       "      <td>BBS9-related Bardet-Biedl syndrome</td>\n",
       "      <td>615986</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biallelic_autosomal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>absent gene product</td>\n",
       "      <td>NaN</td>\n",
       "      <td>loss of function</td>\n",
       "      <td>inferred</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0007707; HP:0001773; HP:0001162; HP:0000750...</td>\n",
       "      <td>16380913; 22353939</td>\n",
       "      <td>DD; Eye</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2025-01-21 22:09:55+00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>G2P00010</td>\n",
       "      <td>TRIM32</td>\n",
       "      <td>602290.0</td>\n",
       "      <td>16380</td>\n",
       "      <td>BBS11; HT2A; LGMD2H; TATIP</td>\n",
       "      <td>TRIM32-related Bardet-Biedl syndrome</td>\n",
       "      <td>615988</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biallelic_autosomal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>uncertain</td>\n",
       "      <td>NaN</td>\n",
       "      <td>undetermined</td>\n",
       "      <td>inferred</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0007707; HP:0001773; HP:0001162; HP:0000750...</td>\n",
       "      <td>16606853</td>\n",
       "      <td>DD; Eye</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2025-01-21 22:12:35+00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>G2P00011</td>\n",
       "      <td>PAX2</td>\n",
       "      <td>167409.0</td>\n",
       "      <td>8616</td>\n",
       "      <td>PAX-2</td>\n",
       "      <td>PAX2-related papillorenal syndrome</td>\n",
       "      <td>120330</td>\n",
       "      <td>NaN</td>\n",
       "      <td>monoallelic_autosomal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>absent gene product</td>\n",
       "      <td>NaN</td>\n",
       "      <td>loss of function</td>\n",
       "      <td>inferred</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0000480; HP:0000588; HP:0002171; HP:0001144...</td>\n",
       "      <td>9760197; 3377002; 11461952; 2644560; 7795640; ...</td>\n",
       "      <td>DD; Eye</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2015-07-22 16:14:08+00:00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     g2p id gene symbol  gene mim  hgnc id       previous gene symbols  \\\n",
       "0  G2P00001        HMX1  142992.0     5017                  H6; NKX5-3   \n",
       "1  G2P00008        YAP1  606608.0    16262                YAP-1; YAP65   \n",
       "2  G2P00009        BBS9  607968.0    30000                   B1; PTHB1   \n",
       "3  G2P00010      TRIM32  602290.0    16380  BBS11; HT2A; LGMD2H; TATIP   \n",
       "4  G2P00011        PAX2  167409.0     8616                       PAX-2   \n",
       "\n",
       "                                        disease name disease mim  \\\n",
       "0               HMX1-related oculoauricular syndrome      612109   \n",
       "1  YAP1-related coloboma, ocular, with or without...      120433   \n",
       "2                 BBS9-related Bardet-Biedl syndrome      615986   \n",
       "3               TRIM32-related Bardet-Biedl syndrome      615988   \n",
       "4                 PAX2-related papillorenal syndrome      120330   \n",
       "\n",
       "  disease MONDO    allelic requirement cross cutting modifier  ...  \\\n",
       "0           NaN    biallelic_autosomal                    NaN  ...   \n",
       "1           NaN  monoallelic_autosomal                    NaN  ...   \n",
       "2           NaN    biallelic_autosomal                    NaN  ...   \n",
       "3           NaN    biallelic_autosomal                    NaN  ...   \n",
       "4           NaN  monoallelic_autosomal                    NaN  ...   \n",
       "\n",
       "   variant consequence variant types molecular mechanism  \\\n",
       "0  absent gene product           NaN    loss of function   \n",
       "1  absent gene product           NaN    loss of function   \n",
       "2  absent gene product           NaN    loss of function   \n",
       "3            uncertain           NaN        undetermined   \n",
       "4  absent gene product           NaN    loss of function   \n",
       "\n",
       "  molecular mechanism categorisation molecular mechanism evidence  \\\n",
       "0                           inferred                          NaN   \n",
       "1                           inferred                          NaN   \n",
       "2                           inferred                          NaN   \n",
       "3                           inferred                          NaN   \n",
       "4                           inferred                          NaN   \n",
       "\n",
       "                                          phenotypes  \\\n",
       "0  HP:0000647; HP:0004328; HP:0000639; HP:0000482...   \n",
       "1  HP:0000407; HP:0003829; HP:0002006; HP:0001249...   \n",
       "2  HP:0007707; HP:0001773; HP:0001162; HP:0000750...   \n",
       "3  HP:0007707; HP:0001773; HP:0001162; HP:0000750...   \n",
       "4  HP:0000480; HP:0000588; HP:0002171; HP:0001144...   \n",
       "\n",
       "                                        publications    panel comments  \\\n",
       "0                       18423520; 25574057; 29140751  DD; Eye      NaN   \n",
       "1                                 24462371; 27267789  DD; Eye      NaN   \n",
       "2                                 16380913; 22353939  DD; Eye      NaN   \n",
       "3                                           16606853  DD; Eye      NaN   \n",
       "4  9760197; 3377002; 11461952; 2644560; 7795640; ...  DD; Eye      NaN   \n",
       "\n",
       "         date of last review  \n",
       "0  2019-09-26 16:23:46+00:00  \n",
       "1  2017-05-12 10:48:00+00:00  \n",
       "2  2025-01-21 22:09:55+00:00  \n",
       "3  2025-01-21 22:12:35+00:00  \n",
       "4  2015-07-22 16:14:08+00:00  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Section 1 -> Loading data\n",
    "\n",
    "df = pd.read_csv(\"G2P_Eye_2025-03-08.csv\")\n",
    "\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "0417a89a-09b4-45b4-ae4e-e4b495d4a40b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Section 2 -> Key/Column mapping.\n",
    "\n",
    "mechanism_map = {\n",
    "    \"loss of function\": (1, 0, 0),\n",
    "    \"gain of function\": (0, 1, 0),\n",
    "    \"dominant negative\": (0, 0, 1),\n",
    "    \"undetermined\": (0, 0, 0)  # Default: No specific mechanism\n",
    "}\n",
    "\n",
    "confidence_map = {\n",
    "    \"definitive\": 5,\n",
    "    \"strong\": 4,\n",
    "    \"moderate\": 3,\n",
    "    \"limited\": 2,\n",
    "    \"disputed\": 1,\n",
    "    \"refuted\": 0\n",
    "}\n",
    "\n",
    "allelic_requirement_map = {\n",
    "    \"biallelic_autosomal\": \"AR\",\n",
    "    \"monoallelic_autosomal\": \"AD\",\n",
    "    \"monoallelic_X\": \"X-linked\",\n",
    "    \"monoallelic_Y_hemizygous\": \"Y-linked\",\n",
    "    \"mitochondrial\": \"mtDNA\"\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "601162ec-27b5-4eb9-ab41-54f49acfba64",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Section 3 -> Helper functions.\n",
    "\n",
    "def parse_mechanism_cat(mech_str):\n",
    "    if pd.isna(mech_str):\n",
    "        return (0, 0, 0)\n",
    "    return mechanism_map.get(mech_str.strip().lower(), (0, 0, 0))\n",
    "\n",
    "def parse_confidence(conf_str):\n",
    "    if pd.isna(conf_str):\n",
    "        return 0\n",
    "    return confidence_map.get(conf_str.strip().lower(), 0)\n",
    "\n",
    "def parse_allelic_requirement(ar_str):\n",
    "    if pd.isna(ar_str):\n",
    "        return \"\"\n",
    "    return allelic_requirement_map.get(ar_str.strip(), ar_str)\n",
    "\n",
    "def log_normalize(series):\n",
    "    if series.max() == 0:\n",
    "        return series  \n",
    "    return np.log1p(series) / np.log1p(series.max())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "d5a06d62-6cd7-42cd-85c1-d7efe4b35e9a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Section 4 -> DF transformation.\n",
    "\n",
    "df[\"LoF\"], df[\"GoF\"], df[\"DN\"] = zip(*df[\"molecular mechanism\"].apply(parse_mechanism_cat))\n",
    "\n",
    "df[\"ConfidenceScore\"] = df[\"confidence\"].apply(parse_confidence) if \"confidence\" in df.columns else 0\n",
    "\n",
    "df[\"Mechanism_Vector_4D\"] = df.apply(lambda row: np.array([\n",
    "    row[\"LoF\"],\n",
    "    row[\"GoF\"],\n",
    "    row[\"DN\"],\n",
    "    row[\"ConfidenceScore\"]\n",
    "]), axis=1)\n",
    "\n",
    "if \"allelic requirement\" in df.columns:\n",
    "    df[\"AllelicReq_Simplified\"] = df[\"allelic requirement\"].apply(parse_allelic_requirement)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "771bb72f-da55-49bd-838e-daf6d353c7d7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene symbol</th>\n",
       "      <th>disease name</th>\n",
       "      <th>molecular mechanism</th>\n",
       "      <th>LoF</th>\n",
       "      <th>GoF</th>\n",
       "      <th>DN</th>\n",
       "      <th>ConfidenceScore</th>\n",
       "      <th>Mechanism_Vector_4D</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HMX1</td>\n",
       "      <td>HMX1-related oculoauricular syndrome</td>\n",
       "      <td>loss of function</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>[1, 0, 0, 5]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>YAP1</td>\n",
       "      <td>YAP1-related coloboma, ocular, with or without...</td>\n",
       "      <td>loss of function</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>[1, 0, 0, 4]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BBS9</td>\n",
       "      <td>BBS9-related Bardet-Biedl syndrome</td>\n",
       "      <td>loss of function</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>[1, 0, 0, 5]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRIM32</td>\n",
       "      <td>TRIM32-related Bardet-Biedl syndrome</td>\n",
       "      <td>undetermined</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>[0, 0, 0, 2]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PAX2</td>\n",
       "      <td>PAX2-related papillorenal syndrome</td>\n",
       "      <td>loss of function</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>[1, 0, 0, 5]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>FGFR3</td>\n",
       "      <td>FGFR3-related lacrimo-auriculo-dento-digital s...</td>\n",
       "      <td>undetermined</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>[0, 0, 0, 5]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>FLNB</td>\n",
       "      <td>FLNB-related spondylocarpotarsal synostosis sy...</td>\n",
       "      <td>loss of function</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>[1, 0, 0, 5]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AGPS</td>\n",
       "      <td>AGPS-related rhizomelic chondrodysplasia punctata</td>\n",
       "      <td>undetermined</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>[0, 0, 0, 5]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>RPGRIP1</td>\n",
       "      <td>RPGRIP1-related Leber congenital amaurosis</td>\n",
       "      <td>loss of function</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>[1, 0, 0, 5]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>LAMA1</td>\n",
       "      <td>LAMA1-related cerebellar dysplasia with cysts ...</td>\n",
       "      <td>loss of function</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>[1, 0, 0, 5]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>FTL</td>\n",
       "      <td>FTL-related hereditary hyperferritinemia-catar...</td>\n",
       "      <td>loss of function</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>[1, 0, 0, 5]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>COL2A1</td>\n",
       "      <td>COL2A1-related Kniest dysplasia</td>\n",
       "      <td>dominant negative</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>[0, 0, 1, 5]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GJA1</td>\n",
       "      <td>GJA1-related Hallermann-Streiff syndrome</td>\n",
       "      <td>undetermined</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>[0, 0, 0, 2]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PLK4</td>\n",
       "      <td>PLK4-related microcephaly, growth failure and ...</td>\n",
       "      <td>loss of function</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>[1, 0, 0, 5]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>CLN3</td>\n",
       "      <td>CLN3-related neuronal ceroid lipofuscinosis</td>\n",
       "      <td>loss of function</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>[1, 0, 0, 5]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>FOXE3</td>\n",
       "      <td>FOXE3-related congenital primary aphakia</td>\n",
       "      <td>loss of function</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>[1, 0, 0, 5]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>CLN8</td>\n",
       "      <td>CLN8-related neuronal ceroid lipofuscinosis</td>\n",
       "      <td>loss of function</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>[1, 0, 0, 5]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>MKKS</td>\n",
       "      <td>MKKS-related Mckusick-Kaufman syndrome</td>\n",
       "      <td>loss of function</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>[1, 0, 0, 5]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>LRPAP1</td>\n",
       "      <td>LRPAP1-related myopia, extreme</td>\n",
       "      <td>loss of function</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>[1, 0, 0, 5]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SLC4A11</td>\n",
       "      <td>SLC4A11-related corneal endothelial dystrophy ...</td>\n",
       "      <td>loss of function</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>[1, 0, 0, 4]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   gene symbol                                       disease name  \\\n",
       "0         HMX1               HMX1-related oculoauricular syndrome   \n",
       "1         YAP1  YAP1-related coloboma, ocular, with or without...   \n",
       "2         BBS9                 BBS9-related Bardet-Biedl syndrome   \n",
       "3       TRIM32               TRIM32-related Bardet-Biedl syndrome   \n",
       "4         PAX2                 PAX2-related papillorenal syndrome   \n",
       "5        FGFR3  FGFR3-related lacrimo-auriculo-dento-digital s...   \n",
       "6         FLNB  FLNB-related spondylocarpotarsal synostosis sy...   \n",
       "7         AGPS  AGPS-related rhizomelic chondrodysplasia punctata   \n",
       "8      RPGRIP1         RPGRIP1-related Leber congenital amaurosis   \n",
       "9        LAMA1  LAMA1-related cerebellar dysplasia with cysts ...   \n",
       "10         FTL  FTL-related hereditary hyperferritinemia-catar...   \n",
       "11      COL2A1                    COL2A1-related Kniest dysplasia   \n",
       "12        GJA1           GJA1-related Hallermann-Streiff syndrome   \n",
       "13        PLK4  PLK4-related microcephaly, growth failure and ...   \n",
       "14        CLN3        CLN3-related neuronal ceroid lipofuscinosis   \n",
       "15       FOXE3           FOXE3-related congenital primary aphakia   \n",
       "16        CLN8        CLN8-related neuronal ceroid lipofuscinosis   \n",
       "17        MKKS             MKKS-related Mckusick-Kaufman syndrome   \n",
       "18      LRPAP1                     LRPAP1-related myopia, extreme   \n",
       "19     SLC4A11  SLC4A11-related corneal endothelial dystrophy ...   \n",
       "\n",
       "   molecular mechanism  LoF  GoF  DN  ConfidenceScore Mechanism_Vector_4D  \n",
       "0     loss of function    1    0   0                5        [1, 0, 0, 5]  \n",
       "1     loss of function    1    0   0                4        [1, 0, 0, 4]  \n",
       "2     loss of function    1    0   0                5        [1, 0, 0, 5]  \n",
       "3         undetermined    0    0   0                2        [0, 0, 0, 2]  \n",
       "4     loss of function    1    0   0                5        [1, 0, 0, 5]  \n",
       "5         undetermined    0    0   0                5        [0, 0, 0, 5]  \n",
       "6     loss of function    1    0   0                5        [1, 0, 0, 5]  \n",
       "7         undetermined    0    0   0                5        [0, 0, 0, 5]  \n",
       "8     loss of function    1    0   0                5        [1, 0, 0, 5]  \n",
       "9     loss of function    1    0   0                5        [1, 0, 0, 5]  \n",
       "10    loss of function    1    0   0                5        [1, 0, 0, 5]  \n",
       "11   dominant negative    0    0   1                5        [0, 0, 1, 5]  \n",
       "12        undetermined    0    0   0                2        [0, 0, 0, 2]  \n",
       "13    loss of function    1    0   0                5        [1, 0, 0, 5]  \n",
       "14    loss of function    1    0   0                5        [1, 0, 0, 5]  \n",
       "15    loss of function    1    0   0                5        [1, 0, 0, 5]  \n",
       "16    loss of function    1    0   0                5        [1, 0, 0, 5]  \n",
       "17    loss of function    1    0   0                5        [1, 0, 0, 5]  \n",
       "18    loss of function    1    0   0                5        [1, 0, 0, 5]  \n",
       "19    loss of function    1    0   0                4        [1, 0, 0, 4]  "
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Section 5 -> Result and export.\n",
    "\n",
    "df[[\n",
    "    \"gene symbol\",\n",
    "    \"disease name\",\n",
    "    \"molecular mechanism\",\n",
    "    \"LoF\", \"GoF\", \"DN\",\n",
    "    \"ConfidenceScore\",\n",
    "    \"Mechanism_Vector_4D\"\n",
    "]].head(20)\n",
    "\n",
    "# df.to_csv(\"G2P_annotated.csv\", index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:base] *",
   "language": "python",
   "name": "conda-base-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
